File: led/led_BoardRev2.0/led_python/led.py
# -- coding: utf-8 --
#导入模块
import RPi.GPIO as GPIO
import time
#定义一个存放led引脚号的列表
leds_pin = [23,27,22,5];

def led_init():
#设置引脚模式为BCM引脚号模式
    GPIO.setmode(GPIO.BCM);
#循环遍历存放引脚的数组
    for x in leds_pin:
        GPIO.setup(x,GPIO.OUT);
        pass
#定义点亮led的函数on(i),参数为led号，编号范围为1~4,低电平点亮
def on(i):
    if(i>=1 and i<=4):
        GPIO.output(leds_pin[i-1],GPIO.LOW);
#定义熄灭led的函数on(i),参数为led号，编号范围为1~4
def off(i):
    if(i>=1 and i<=4):
        GPIO.output(leds_pin[i-1],GPIO.HIGH);
#定义流水灯函数flow(time),time为流水速度，单位为秒
def flow(delay):
    for i in range(len(leds_pin)):
        GPIO.output(leds_pin[i],GPIO.LOW);
        print('LED',i+1,'is on');
        time.sleep(delay);
        GPIO.output(leds_pin[i],GPIO.HIGH);
        print('LED',i+1,'is off');
        time.sleep(delay);
#GPIO清理函数，调用后释放GPIO
def clean():
    GPIO.cleanup();

File: led/led_BoardRev2.0/led_wiringPi/led.c
/**************************************************************
	author:MrzhangF1ghter 
	Last Edit time:2018-4-30
	此为RainbowBoard LED例程 使用wiringPi库
	彩虹板上led灯物理连接如下
	灯   | GPIO | wPi |排针号|
	LED1 BCM23->pin4	16    
	LED2 BCM27->pin2	13
	LED3 BCM22->pin3	15
	LED4 BCM5 ->pin21	29
**************************************************************/
#include<stdio.h>
#include<unistd.h>
#include<wiringPi.h>//wiringPi路径
int main()
{
		int leds_pin[4]={4,2,3,21};//定义一个存放led对应gpio引脚号的整形数组
		int i;//定义一个用于循环计数变量
		printf("Welcome to IODevelopBoard!\n");//打印欢迎信息
		printf("LEDS test,wiringPi version\n");//打印欢迎信息
		wiringPiSetup();//wiringPi库初始化
		/*用循环一条语句设置完所有LED引脚为OUTPUT模式，
		函数原型digitalWrite (int pin, int value)*/
		for(i=0;i<4;i++)
		{
			pinMode(leds_pin[i],OUTPUT);//设置led_pin[i]引脚为输出模式
			digitalWrite(leds_pin[i],HIGH);//初始化为高电平
		}
		
		while(1)//流水灯
		{
			for(i=0;i<4;i++)
			{
				digitalWrite(leds_pin[i],LOW);//LED1引脚低电平（低电平点亮）
				printf("LED%d is on!\n",i+1);
				sleep(1);//程序休眠1秒
				digitalWrite(leds_pin[i],HIGH);//LED1引脚高电平（高电平熄灭）
				printf("LED%d is off!\n",i+1);
				sleep(1);//程序休眠1秒
			}
		}
}
